src/d/actor/d_a_obj_ospbox.cpp

```diff
@@ -4,77 +4,229 @@
+        /* Height */ 150.0f,
+    },
+};
 
 /* 000000EC-000001A0       .text CreateHeap__Q211daObjOspbox5Act_cFv */
 BOOL daObjOspbox::Act_c::CreateHeap() {
-    /* Nonmatching */
+    J3DModelData* model_data = (J3DModelData*)dComIfG_getObjectRes(M_arcname, OSPBOX_BDL_OSPBOX);
+    JUT_ASSERT(0x86, model_data != NULL);
+    mpModel = mDoExt_J3DModel__create(model_data,0x80000,0x11000022);
+    return mpModel != NULL;
 }
 
 /* 000001A0-00000294       .text Create__Q211daObjOspbox5Act_cFv */
 BOOL daObjOspbox::Act_c::Create() {
-    /* Nonmatching */
+    fopAcM_SetMtx(this, mpModel->getBaseTRMtx());
+    init_mtx();
+
+    fopAcM_setCullSizeBox(this, -76.0f, -1.0f, -76.0f, 76.0f, 151.0f, 76.0f);
+
+    mStts.Init(0xFF, 0xFF, this);
+    mCyl.Set(M_cyl_src);
+    mCyl.SetStts(&mStts);
+    mCyl.SetTgVec((cXyz&)cXyz::Zero);
+    mCyl.OnTgNoHitMark();
+
+    attention_info.position.y = fopAcM_GetPosition_p(this)->y + 75.0f;
+    eyePos = attention_info.position;
+
+    init_ground();
+
+    m49A = 2;
+    return TRUE;
 }
 
+const s16 pf_name[] = {0x56, 0xDD, 0xDD, 0xDD, 0x56, 0x56, 0x56, 0x56}; // TODO: double-check
+const u32 prm[] = {0x00000000, 0x01FF0000, 0x01FF0001, 0x01FF0002}; // TODO: double-check
+
+Mtx daObjOspbox::Act_c::M_tmp_mtx;
+
 /* 00000294-000004F8       .text Mthd_Create__Q211daObjOspbox5Act_cFv */
 cPhs_State daObjOspbox::Act_c::Mthd_Create() {
-    /* Nonmatching */
+    fopAcM_SetupActor(this, Act_c);
+    cPhs_State phase_state = dComIfG_resLoad(&mPhase, M_arcname);
+    if (phase_state == cPhs_COMPLEATE_e) {
+        phase_state = MoveBGCreate(M_arcname, OSPBOX_DZB_OSPBOX, NULL, 0x8a0);
+        JUT_ASSERT(0xc3, (phase_state == cPhs_COMPLEATE_e) || (phase_state == cPhs_ERROR_e));
+    }
+    return phase_state;
 }
 
 /* 00000BAC-00000BB4       .text Delete__Q211daObjOspbox5Act_cFv */
 BOOL daObjOspbox::Act_c::Delete() {
-    /* Nonmatching */
+    return TRUE;
 }
 
 /* 00000BB4-00000C04       .text Mthd_Delete__Q211daObjOspbox5Act_cFv */
 BOOL daObjOspbox::Act_c::Mthd_Delete() {
-    /* Nonmatching */
+    BOOL result = MoveBGDelete();
+    dComIfG_deleteObjectRes(M_arcname);
+    return result;
 }
 
 /* 00000C04-00000C84       .text set_mtx__Q211daObjOspbox5Act_cFv */
 void daObjOspbox::Act_c::set_mtx() {
-    /* Nonmatching */
+    mDoMtx_stack_c::transS(current.pos.x, current.pos.y, current.pos.z);
+    mDoMtx_ZXYrotM(mDoMtx_stack_c::get(), shape_angle.x, shape_angle.y, shape_angle.z);
+    mpModel->setBaseTRMtx(mDoMtx_stack_c::get());
+    PSMTXCopy(mDoMtx_stack_c::get(), M_tmp_mtx);
 }
 
 /* 00000C84-00000CC0       .text init_mtx__Q211daObjOspbox5Act_cFv */
 void daObjOspbox::Act_c::init_mtx() {
-    /* Nonmatching */
+    mpModel->setBaseScale(scale);
+    set_mtx();
 }
 
 /* 00000CC0-00000D7C       .text make_item__Q211daObjOspbox5Act_cFv */
 void daObjOspbox::Act_c::make_item() {
-    /* Nonmatching */
+    int prm_index = prm_get_spec();
+    if (pf_name[prm_index] == 0x56) {
+        prm_index = prm_get_itemNo();
```

Comment:
```suggestion
        int itemNo = prm_get_itemNo();
```
Reusing the var doesn't make sense here